### PR TITLE
Add a CONTRIBUTING.md file

### PR DESCRIPTION
**Pull request by ppmadalin**
I have started a contribution file.

### CODE REVIEW

**Comment by code-monk08** [comment]
hey @ppmadalin, I see a lot of typos in there, can you fix them? :)

**Comment by ppmadalin** [comment]
Yep! Sorry for that, it was around 1 AM when I wrote this. 😄

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Welcome to OpenTrivia project!
+
+We are glad to be hear. So? You want to contribute to our project? Great,
```
Comment (ppmadalin) [inline CONTRIBUTING.md]: here not her. One is spotted

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+bring here your best ideas. We are looking forward to work along with You 
+and to have fun by making this project better. 
+
+## What PR we accept?
+
+Theoretically, everything wich can add value to this project is accepted. That's all!
+However, in order to give them a name, we are looking for the following contribution types:
+
+1. Code Enhancements: Did you spot an ugly code? Let's make it better.  
+2. Ideas: Do you have an idea about a new feature wich we ca add? Let's us know about it. Don't be shine.
```
Comment (ppmadalin) [inline CONTRIBUTING.md]: "Don't be shine!!!? What? 😄

**Comment by code-monk08** [comment]
@ppmadalin I reviewed your PR, please make the necessary changes :)

**Comment by ppmadalin** [comment]
@code-monk08 can you help me a little bit and point me to those changes need to be done? 😄

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,8 +1,13 @@
...
 
-We are glad to be heare. So? You want to contribute to our project? Great, 
+We are glad you are here. So, you want to contribute to our project? Great, 
 bring here your best ideas. We are looking forward to work along with You 
-and to have fun by making this project better. 
+and to have fun together by making this project better. 
+
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make via issue, email,
```
[the remark this answers is not in the document]
Comment (ppmadalin) [reply CONTRIBUTING.md]: Done. 😄

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -13,6 +18,24 @@ However, in order to give them a name, we are looking for the following contribu
 2. Ideas: Do you have an idea about a new feature wich we ca add? Let's us know about it.
```
[the remark this answers is not in the document]
Comment (ppmadalin) [reply CONTRIBUTING.md]: Done. 😄

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,8 +1,13 @@
 # Welcome to OpenTrivia project!
 
-We are glad to be heare. So? You want to contribute to our project? Great, 
+We are glad you are here. So, you want to contribute to our project? Great, 
 bring here your best ideas. We are looking forward to work along with You
```
[the remark this answers is not in the document]
Comment (ppmadalin) [reply CONTRIBUTING.md]: Done. 😄

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -31,11 +31,11 @@ However, in order to give them a name, we are looking for the following contribu
...
 	upstream
 
-	# Now, because we want our master to be up to date with the original forked 
-	# master, we have to set this using the below command.
+	# Now, because we want our master branch to be up to date with the forked branch 
+	# we need to set this using the below command.
 	$ git branch --set-upstream-to=upstream/master master
 	
-	# When you doing a pull, the changes will not gone comming from 
+	# When you are doing a pull, the changes will not gone coming from
```
[the remark this answers is not in the document]
Comment (ppmadalin) [reply CONTRIBUTING.md]: Done. 😄

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+	# we need to set this using the below command.
+	$ git branch --set-upstream-to=upstream/master master
+	
+	# When you are doing a pull, the changes will not gone coming from 
+	# origin/master, instead they will come from upstream/master.
+	git pull
+	```
+4. Create a branch
+	`$ git checkout -b <your-branch-name>`
+5. Before starting doing something, run the tests and ensure yourself that
```
[the remark this answers is not in the document]
Comment (ppmadalin) [reply CONTRIBUTING.md]: Done. 😄

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+	# When you are doing a pull, the changes will not gone coming from 
+	# origin/master, instead they will come from upstream/master.
+	git pull
+	```
+4. Create a branch
+	`$ git checkout -b <your-branch-name>`
+5. Before starting doing something, run the tests and ensure yourself that 
+   everything works and nothing is broken. If everything works, 
+   then you are ready to make changes. 
+7. Make changes. Run again the tests. If the tests are passing. You are on the right way to push the changes.
```
[the remark this answers is not in the document]
Comment (ppmadalin) [reply CONTRIBUTING.md]: Done. 😄

File: .gitignore [hunk 9]
```diff
@@ -1,2 +1,124 @@
...
+# mkdocs documentation
+/site
+
+# mypy
+.mypy_cache/
+.dmypy.json
+dmypy.json
+
+# Pyre type checker
+.pyre/
```
[the remark this answers is not in the document]
Comment (ppmadalin) [reply .gitignore]: Not sure what Instrucitons.md is 🤔

Is it a different file we need to create and add a link of it to Contribution?

[the remark this answers is not in the document]
Comment (code-monk08) [reply .gitignore]: oh actually it's a temporary file which I created for myself..I have actually added most of instructions in current readme.

[the remark this answers is not in the document]
Comment (code-monk08) [reply .gitignore]: I will add `instrcutions.md` to the gitignore file, I am merging this PR, thanks for the amazing work. :)

**Comment by code-monk08** [comment]
I will add instrcutions.md to the gitignore file, I am merging this PR, thanks for the amazing work. :)

**Pull request merged**